### PR TITLE
fix map for too many boxes

### PR DESCRIPTION
**Pull request by graft**
Sometimes if there are too many boxes in a row on the map they overlap. This uses a max-width to make sure that doesn't happen.

### CODE REVIEW

**Comment by graft** [comment]
Peek 2023-02-16 11-43

File: timur/lib/client/jsx/components/model_map/model_node.jsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-const ModelNode = ({model_name, center, size, selected, handler, disabled}) =>
+const ModelNode = ({model_name, center, size, selected, handler, disabled, numSiblings}) =>
+
   center ? (
     <div
       className={`model_node ${selected ? 'selected' : ''} ${
         disabled ? 'disabled' : ''
       }`}
-      style={{top: center.y, left: center.x}}
+      style={{top: center.y, left: center.x, '--model-width': `${600 / (numSiblings + 1) - 15}px` }}
```
Comment (graft) [inline timur/lib/client/jsx/components/model_map/model_node.jsx:10]: This is a pretty cool trick (paired with `var(--model-width)` in the css) that lets you pass data into any css styles set on this thing without having to override the style itself (so precedence isn't destroyed).

Comment (coleshaw) [reply timur/lib/client/jsx/components/model_map/model_node.jsx:10]: oh neat. Is this a React thing, where you can pass in / use CSS variables?

File: timur/lib/client/jsx/components/model_map/tree_layout.jsx [hunk 2]
```diff
@@ -87,6 +87,8 @@ class TreeLayout {
 
     let grid = [];
 
+    this.grid = grid;
```
Comment (coleshaw) [inline timur/lib/client/jsx/components/model_map/tree_layout.jsx:90]: Do you still need this?

**Review by coleshaw — APPROVED** [review]
Nice, this looks like a pretty clean approach to addressing the issue!

File: timur/lib/client/jsx/components/model_map/model_node.jsx (hunk 1 above)
Comment (dtm2451) [reply timur/lib/client/jsx/components/model_map/model_node.jsx:10]: pretty cool!

File: timur/lib/client/jsx/components/model_map/tree_layout.jsx (hunk 2 above)
Comment (graft) [reply timur/lib/client/jsx/components/model_map/tree_layout.jsx:90]: Yeah has to be retained to figure out how many siblings each node has.

File: timur/lib/client/jsx/components/model_map/model_node.jsx (hunk 1 above)
Comment (graft) [reply timur/lib/client/jsx/components/model_map/model_node.jsx:10]: Apparently, pure CSS/html.

**Review by dtm2451 — APPROVED** [review]
I like this approach a lot!  IMO, it works well enough even when there are MANY models, like the 15 in this gif =)
map_edit_model_names

**Pull request merged**